From a318f4c6708246229de62712cc82d6075b1f6907 Mon Sep 17 00:00:00 2001
From: Christian Ege <k4230r6@gmail.com>
Date: Sat, 15 Oct 2016 21:19:48 +0200
Subject: [PATCH] Added env variable handling like UDOO u-boot

This patch adds support for the same bootscript type like the UDOO boards
do use. The bootloader tries to load a uEnv.txt from mmc and import those
environment variables. This can be used to overwrite append the u-boot
environment.

Signed-off-by: Christian Ege <k4230r6@gmail.com>
---
 include/configs/udoo.h | 19 +++++++++----------
 1 file changed, 9 insertions(+), 10 deletions(-)

diff --git a/include/configs/udoo.h b/include/configs/udoo.h
index fba78d0..993f29f 100644
--- a/include/configs/udoo.h
+++ b/include/configs/udoo.h
@@ -63,7 +63,7 @@
 #define CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
-	"script=boot.scr\0" \
+	"script=uEnv.txt\0" \
 	"image=zImage\0" \
 	"console=ttymxc1\0" \
 	"splashpos=m,m\0" \
@@ -75,6 +75,7 @@
 	"ip_dyn=yes\0" \
 	"mmcdev=0\0" \
 	"mmcpart=1\0" \
+        "video_output=hdmi\0" \
 	"mmcroot=/dev/mmcblk0p2 rootwait rw\0" \
 	"update_sd_firmware_filename=u-boot.imx\0" \
 	"update_sd_firmware=" \
@@ -91,11 +92,11 @@
 			"fi; "	\
 		"fi\0" \
 	"mmcargs=setenv bootargs console=${console},${baudrate} " \
-		"root=${mmcroot}\0" \
+		"root=${mmcroot} ${video}\0" \
 	"loadbootscript=" \
 		"load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
-		"source\0" \
+		"env import -t ${loadaddr} ${filesize};\0" \
 	"loadimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
 	"loadfdt=load mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
@@ -139,9 +140,9 @@
 		"fi;\0" \
 		"findfdt=" \
 			"if test $board_rev = MX6Q ; then " \
-				"setenv fdt_file imx6q-udoo.dtb; fi; " \
+				"setenv fdt_file imx6q-udoo-${video_output}.dtb; fi; " \
 			"if test $board_rev = MX6DL ; then " \
-				"setenv fdt_file imx6dl-udoo.dtb; fi; " \
+				"setenv fdt_file imx6dl-udoo-${video_output}.dtb; fi; " \
 			"if test $fdt_file = undefined; then " \
 				"echo WARNING: Could not determine dtb to use; fi; \0"
 
@@ -150,12 +151,10 @@
 	   "mmc dev ${mmcdev}; if mmc rescan; then " \
 		   "if run loadbootscript; then " \
 			   "run bootscript; " \
-		   "else " \
-			   "if run loadimage; then " \
-				   "run mmcboot; " \
-			   "else run netboot; " \
-			   "fi; " \
 		   "fi; " \
+		    "if run loadimage; then " \
+			"run mmcboot; " \
+ 		   "fi; " \
 	   "else run netboot; fi"
 
 /* Print Buffer Size */
-- 
2.1.4

